src/opm/parser/eclipse/EclipseState/checkDeck.cpp

```diff
@@ -48,6 +48,49 @@ bool checkDeck( const Deck& deck, const Parser& parser, size_t enabledChecks) {
         deckValid = deckValid && Section::checkSectionTopology(deck, parser, ensureKeywordSection);
     }
 
+    // ensure that only a single unit system is used by the deck
+    int numUnitSystems = 0;
+    if (deck.hasKeyword("FIELD"))
+        ++ numUnitSystems;
+    if (deck.hasKeyword("METRIC"))
+        ++ numUnitSystems;
+    if (deck.hasKeyword("LAB"))
+        ++ numUnitSystems;
+    if (deck.hasKeyword("PVT-M"))
+        ++ numUnitSystems;
+
+    if (numUnitSystems != 1) {
+        std::string msg("Exactly one unit system must be specified by the deck (is: " + std::to_string(numUnitSystems) + ").");
+        OpmLog::warning(msg);
+        deckValid = false;
+    }
+
+    // make sure that no FILEUNIT keyword that specifies a different unit system as the
+    // deck is present
+    for (size_t kwIdx = 0; kwIdx < deck.size(); ++ kwIdx) {
```

Comment:
This loop can be simplified:
```C++
const std::string& unit_system = deck.getActiveUnitSystem().getName();
for (const auto& kw_ptr : deck.getKeywordList("FILEUNIT")) {
     if (kw_ptr->getRecord(0).getItem("FILE_UNIT_SYSTEM").getTrimmedString(0) != unit_system) {
            std::string msg = "Unit system f... up at " + kw_ptr->getFileName() + ":" + std::to_string(kw_ptr->getLineNumber());
             parseContext.handleError(UNIT_SYSTEM_FUCK_UP, msg , errors);  
     }
}
```

Reply:
I picked as simpler implementation to fix this. if this turns out to be a common mistake (I doubt it), the error messages should get spiced up and something like that is needed...